pkg/webhook/containerrecreaterequest/validating/crr_create_update_handler.go

```diff
@@ -0,0 +1,62 @@
+package validating
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/openkruise/kruise/pkg/features"
+	"github.com/openkruise/kruise/pkg/util/controllerfinder"
+	utilfeature "github.com/openkruise/kruise/pkg/util/feature"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// ContainerRecreateRequestHandler handles ContainerRecreateRequest
+type ContainerRecreateRequestHandler struct {
+	Client client.Client
+	// Decoder decodes objects
+	Decoder *admission.Decoder
+	finders *controllerfinder.ControllerFinder
+}
+
+var _ admission.Handler = &ContainerRecreateRequestHandler{}
+
+func (h *ContainerRecreateRequestHandler) validatingContainerRecreateRequestFn(ctx context.Context, req admission.Request) (allowed bool, reason string, err error) {
+	allowed = true
+
+	switch req.Operation {
+	case admissionv1.Create:
+		if utilfeature.DefaultFeatureGate.Enabled(features.PodUnavailableBudgetDeleteGate) {
```

Comment:
Maybe we should consider container restart as an Update operation instead of Delete, which means you should check `PodUnavailableBudgetUpdateGate` feature-gate.